cucumber-expressions/javascript/src/cucumber_expression.js

```diff
@@ -7,7 +7,7 @@ class CucumberExpression {
    * @param parameterTypeRegistry
    */
   constructor (expression, types, parameterTypeRegistry) {
-    const PARAMETER_REGEXP = /\{([^}:]+)(:([^}]+))?}/g
+      var PARAMETER_REGEXP = /[{\[]([^}\]:]+)(:([^}\]]+))?[}\]]/g;
```

Comment:
I think it would be better to make the opening `[{\[]` and closing `[}\]]` capture groups. That would simplify the code below where `matchType` is assigned.

Formatting:
* Indent with 2 spaces
* Use `const`, not `var`
* No semicolons